gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -190,12 +189,13 @@ discreteElimination(const HybridGaussianFactorGraph &factors,
 // If any GaussianFactorGraph in the decision tree contains a nullptr, convert
 // that leaf to an empty GaussianFactorGraph. Needed since the DecisionTree will
 // otherwise create a GFG with a single (null) factor.
+// TODO(dellaert): still a mystery to me why this is needed.
```

Comment:
@dellaert about this TODO, the explanation is given above. The `assembleGraphTree` method causes an empty `GaussianFactorGraph` to be created when there is a nullptr, hence we need to explicitly check and remove any factor graph with nullptrs (by setting it to nullptr).